src/test/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidatorTest.java

```diff
@@ -338,6 +338,24 @@ public void invalidCollection_notObject() throws Exception {
         assertThat(result1.getFieldId(), equalTo("Initials.1"));
     }
 
+    @Test
+    public void shouldFailForTextCaseReference() throws Exception {
+        final String DATA = "{\n" +
+            "        \"CaseReference\": \"1596XXXXX1048-4059XXXOOOO\"\n" +
+            "      }";
+
+        final Map<String, JsonNode> values = MAPPER.readValue(DATA, new TypeReference<HashMap<String, JsonNode>>() {
+        });
+        final List<ValidationResult> results = caseDataValidator.validate(values, caseFields);
+        assertEquals(results.toString(), 1, results.size());
+
+        final ValidationResult result0 = results.get(0);
+        assertThat(result0.getFieldId(), equalTo("CaseReference"));
+        assertThat(result0.getErrorMessage(),
+            equalTo("The data entered is not valid for this type of field, please delete and re-enter using only valid data")
+        );
+    }
```

Comment:
`CaseDataValidatorTest` is a generic class, should not contain tests for a specific field like this test. Le't make it more generic

Reply:
mario, The current test is validating counreate text fields features.  for instance textFieldWithMaxMin  and textFieldWithInvalidMaxMin .  Should a rename my test for refer to a predefined-field ? or Should I Create a new test class for Predefined types ?

Reply:
More test has been added . DONE